### PR TITLE
Auto build dist

### PR DESCRIPTION
**Pull request by danielrbradley**
Push dist changes back to a PR if out of date.

This is a nice opportunity to prototype how this mechanism could work elsewhere too.

### CODE REVIEW

**Comment by danielrbradley** [comment]
One slightly odd behaviour I can't quite explain right now is why the commit that was pushed by pulumi-bot isn't then re-tested.

Perhaps there's something about using the built-in token that github detects and disables triggering workflows for the pushed commit to avoid infinite loops?

File: .github/workflows/check-dist.yml [hunk 1]
```diff
@@ -70,3 +73,14 @@ jobs:
...
+      - if:
+          ${{ failure() && steps.diff.outcome == 'failure' && github.event_name
+          == 'pull_request'}}
+        name: Push updated build
+        run: |
+          git config --global user.name pulumi-bot
+          git config --global user.email bot@pulumi.com
+          git add dist/
+          git commit -m "Rebuild dist"
+          git push
```
Comment (iwahbe) [inline .github/workflows/check-dist.yml:86]: It's not clear to me what this is pushing back to? Doesn't this script run only on `main`? I'd rather not set up a CI job that pushes directly to main.

Comment (danielrbradley) [reply .github/workflows/check-dist.yml:86]: This pushes back to the PR's branch - therefore allowing the failing PR to start passing.

Comment (danielrbradley) [reply .github/workflows/check-dist.yml:86]: If it's not running against a PR it'll skip this step.

Comment (iwahbe) [reply .github/workflows/check-dist.yml:86]: Does this manifest as SDK regeneration? I'm having trouble groking the desired change. `dist` is not meaningful to me.

**Comment by iwahbe** [comment]
> One slightly odd behaviour I can't quite explain right now is why the commit that was pushed by pulumi-bot isn't then re-tested.
> 
> Perhaps there's something about using the built-in token that github detects and disables triggering workflows for the pushed commit to avoid infinite loops?

I think this needs to be fixed before merging. Could the job kick off an explicit workflow run?

File: .github/workflows/check-dist.yml (hunk 1 above)
Comment (danielrbradley) [reply .github/workflows/check-dist.yml:86]: Oh right - nodejs actions have to have compilled code committed to be runnable without extra tooling. This goes into the `dist` folder.

This isn't related to providers, though the same technique could perhaps be applied to providers for re-generating schema & SDKs if it was forgotten to be rebuilt. I think Guin was working on something like this.

**Review by iwahbe — APPROVED** [review]
It took a while to understand what's going on. As I understand now:

When there is a PR to this repo, if the `dist` folder (the transpired TS code) is out of date (failing CI), then CI will push an up to date copy of `dist` back to the PR. That sounds good to me.

**Pull request merged**